### PR TITLE
SDKS-2272 Introduce FRWebAuthn interface to remove WebAuthn Reference Keys 

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[Please link jira ticket here](https://bugster.forgerock.org/jira/browse/SDKS-2272)

# Description

Introduce FRWebAuthn interface to remove WebAuthn Reference Keys

# Definition of Done Checklist:

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility.
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] Functional spec is written/updated.
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/WebAuthnDataRepository.kt [hunk 1]
```diff
@@ -119,20 +119,36 @@ open class WebAuthnDataRepository internal constructor(val context: Context,
...
-            fromJson(JSONObject(it)).userHandle.contentEquals(userId).not()
-        }?.toSet()
+        val storedCredentials =
+            getPublicKeyCredentialSource(publicKeyCredentialSource.rpid).filter {
+                it.id.contentEquals(publicKeyCredentialSource.id).not()
+            }.map { it.toJson().toString() }.toSet()
 
-        storedCredentials?.let {
-            dataRepository.edit().putStringSet(rpId, it).apply()
+        storedCredentials.let {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/WebAuthnDataRepository.kt:137]: dont need this let { } .  Let can be only used for optional. just go straight dataRepository.edit().putStringSet(publicKeyCredentialSource.rpid, storedCredentials).apply()

**Review by jeyanthanperiyasamy — COMMENTED** [review]
just a minor comment ..All good

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * The [PublicKeyCredentialSource] only contains the reference to the actual key,
+ * deleting the [PublicKeyCredentialSource] does not delete the actual key
+ */
+class FRWebAuthn(private val context: Context,
+                 private val webAuthnDataRepository: WebAuthnDataRepository =
+                     WebAuthnDataRepository.WebAuthnDataRepositoryBuilder().context(context)
+                         .build()) {
+
+    /**
+     * Delete the [PublicKeyCredentialSource] by Relying Party Id
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt:24]: can we also add the parameter (rpId) in the Doc ,

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    fun deleteCredentials(rpId: String) {
+        webAuthnDataRepository.delete(rpId)
+    }
+
+    /**
+     * Delete the provide [PublicKeyCredentialSource], the [PublicKeyCredentialSource.id] will
+     * be used as the key to lookup from internal storage
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt:32]: can we also add the parameter (publicKeyCredentialSource) in the Doc ,

**Review by jeyanthanperiyasamy — COMMENTED** [review]
minor update

File: forgerock-auth/src/main/java/org/forgerock/android/auth/WebAuthnDataRepository.kt (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/WebAuthnDataRepository.kt]: Updated

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt (hunk 2 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt:24]: Updated

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt (hunk 3 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/FRWebAuthn.kt:33]: Updated

**Review by jeyanthanperiyasamy — DISMISSED** [review]

**Review by spetrov — APPROVED** [review]
Looks good! 👍🏻

**Pull request merged**